stacks_queues/queue.py

```diff
@@ -23,39 +22,63 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+
+        if self.size == self.buffer_size:
+            raise QueueFullException()
+        
+        if self.front == -1:
+            self.front = self.rear = 0
+
+        # if self.rear == self.buffer_size:
+        #     self.rear = 0
+
+        self.size += 1
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.size == 0:
+            raise QueueEmptyException()
+
+        self.size -= 1
+        element = self.store[self.front]
+        self.front += 1
+        if self.front == self.buffer_size:
+            self.front = 0
+        return element
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.size == 0:
+            return None
+        else:
+            return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        return self.size == 0
 
     def __str__(self):
         """ Returns the Queue in String form like:
             [3, 4, 7]
             Starting with the front of the Queue and
             ending with the rear of the Queue.
         """
-        pass
+        return str(self.store[self.front:self.rear])
```

Comment:
This will only work in the case that the rear pointer has an index greater than that of the front index.

In the case that the rear pointer has wrapped around to the front of the queue, the rear pointer will be less than the front pointer and your code will end up returning an empty string.